pkg/agent/config_data_gatherers.go

```diff
@@ -0,0 +1,159 @@
+		if err != nil {
+			return dg, fmt.Errorf("failed to load AKS DataGatherer: %v", err)
+		}
+		return dg, nil
+	} else if kind == "local" {
+		dg, err := loadLocalConfig(config)
+		if err != nil {
+			return dg, fmt.Errorf("failed to load Local DataGatherer: %v", err)
+		}
+		return dg, nil
+	} else {
+		return dg, fmt.Errorf("cannot load data gatherer, kind '%s' is not supported", kind)
+	}
+}
+
+func loadGKEConfig(ctx context.Context, config map[string]string) (datagatherer.DataGatherer, error) {
+	var result *multierror.Error
+
+	if config["project"] == "" {
+		result = multierror.Append(result, fmt.Errorf("project is required"))
+	}
+	if config["cluster"] == "" {
+		result = multierror.Append(result, fmt.Errorf("cluster is required"))
+	}
+	if config["location"] == "" {
+		result = multierror.Append(result, fmt.Errorf("location is required"))
+	}
+	if config["credentials"] == "" {
+		result = multierror.Append(result, fmt.Errorf("credentials is required"))
+	}
+
+	cluster := gke.Cluster{
+		Project:  config["project"],
+		Name:     config["cluster"],
+		Location: config["location"],
+	}
+
+	return gke.NewGKEDataGatherer(ctx, &cluster, config["credentials"]), result.ErrorOrNil()
+}
+
+func loadK8sConfig(config map[string]string) (datagatherer.DataGatherer, error) {
+	var dg datagatherer.DataGatherer
+
+	if config["kubeconfig"] == "" {
+		return dg, fmt.Errorf("kubeconfig path is required")
+	}
+
+	k8sClient, err := k8s.NewClient(expandHome(config["kubeconfig"]))
+	if err != nil {
+		return dg, fmt.Errorf("failed to create K8s client %s", err)
+	}
+
+	return k8s.NewPodsDataGatherer(k8sClient), nil
+}
+
+func loadEKSConfig(config map[string]string) (datagatherer.DataGatherer, error) {
+	var dg datagatherer.DataGatherer
+
+	if config["cluster"] == "" {
+		return dg, fmt.Errorf("cluster is required")
+	}
+
+	return eks.NewEKSDataGatherer(config["cluster"]), nil
+}
+
+func loadAKSConfig(ctx context.Context, config map[string]string) (datagatherer.DataGatherer, error) {
+	var dg datagatherer.DataGatherer
+	var result *multierror.Error
+
+	if config["resource-group"] == "" {
+		result = multierror.Append(result, fmt.Errorf("project is required"))
+	}
+	if config["cluster"] == "" {
+		result = multierror.Append(result, fmt.Errorf("cluster is required"))
+	}
+	if config["credentials"] == "" {
+		result = multierror.Append(result, fmt.Errorf("location is required"))
+	}
+
+	if result.ErrorOrNil() != nil {
+		return dg, result.ErrorOrNil()
+	}
+
+	return aks.NewAKSDataGatherer(
+		ctx, config["resource-group"], config["cluster"], config["credentials"])
+}
+
+func loadLocalConfig(config map[string]string) (datagatherer.DataGatherer, error) {
+	var dg datagatherer.DataGatherer
+
+	if config["data-path"] == "" {
+		return dg, fmt.Errorf("data-path is required")
+	}
+
+	return local.NewLocalDataGatherer(config["data-path"]), nil
+}
+
+// TODO copied from check, perhaps move to utils or leave here until check has gone from the client?
```

Comment:
https://github.com/jetstack/preflight/pull/102

Reply:
That was a leftover. It is completely gone now.